### PR TITLE
Feat: refactor notification resource code (#280)

### PR DESCRIPTION
**Pull request by TomerHeber**
### Solution

This is the first part of fixing #280. This PR refactors the notification resource a little bit.

### CODE REVIEW

File: client/notification.go [hunk 1]
```diff
@@ -1,5 +1,10 @@
 package client
 
+type NotificationCreatePayloadWith struct {
```
Comment (TomerHeber) [inline client/notification.go:3]: Moved to here, from model.go

File: env0/utils.go [hunk 2]
```diff
@@ -75,7 +75,7 @@ func readResourceData(i interface{}, d *schema.ResourceData) error {
 				field.Set(reflect.ValueOf(sshKeys))
 			}
 		case reflect.String, reflect.Bool, reflect.Int:
-			field.Set(reflect.ValueOf(dval))
+			field.Set(reflect.ValueOf(dval).Convert(fieldType))
```
Comment (TomerHeber) [inline env0/utils.go:78]: This supports "casting" from one type to the other when both types are of the same kind.
E.g.:
```
type MyString string
```

File: env0/utils_test.go [hunk 3]
```diff
@@ -73,3 +71,22 @@ func TestWriteResourceDataModule(t *testing.T) {
 	rawSshKeys = append(rawSshKeys, map[string]interface{}{"id": "id1", "name": "name1"})
 	assert.Equal(t, rawSshKeys, d.Get("ssh_keys"))
 }
+
+func TestReadResourceDataNotification(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (TomerHeber) [reply env0/utils_test.go:75]: https://github.com/env0/terraform-provider-env0/pull/285/commits/f3eca508f9701d3a3a24658d4bbc37a6d2bf1a22

**Review by yaronya — APPROVED** [review]
👏

**Pull request merged**